### DISCUSSION TITLE
Can't mount ZFS dataset

### DISCUSSION THREAD

**Discussion by knacky34** [openzfs/zfs#16003] [Q&A] [answered]
I'm currently facing issue for mounting datasets of an encrypted pool.

The pool and its datasets had been setup like this:
```
# Pool creation
zpool create \
    -o ashift=12 \
    -o autotrim=on \
    -O encryption=on -O keylocation=prompt -O keyformat=passphrase \
    -O acltype=posixacl -O xattr=sa -O dnodesize=auto \
    -O compression=lz4 \
    -O normalization=formD \
    -O atime=off \
    -O canmount=off -O mountpoint=/data -R /mnt \
    dpool raidz1 \
    ${DISK1} ${DISK2} ${DISK3}

# Datasets creation
zfs create -o canmount=off  dpool/nextcloud
zfs create                  dpool/nextcloud/data
``` 

Auto-mounting is not working but I first would like to figure out the issue using manual mounting.
For mounting the datasets, I used :
```
zfs load-key -a
zfs mount dpool/nextcloud/data
```
`zfs mount dpool/nextcloud/data` command exit successfully but the dataset is still not mounted. 
What am I doing wrong ?

**Please note** that altroot is no longer used (export and re-import pool to remove altroot), see:
```
$ zpool list
NAME    SIZE  ALLOC   FREE  CKPOINT  EXPANDSZ   FRAG    CAP  DEDUP    HEALTH  ALTROOT
bpool   960M   105M   855M        -         -     3%    10%  1.00x    ONLINE  -
dpool  8.17T   532G  7.65T        -         -     0%     6%  1.00x    ONLINE  -
rpool   106G  20.0G  86.0G        -         -     3%    18%  1.00x    ONLINE  -
```

**Comment by rincebrain** [comment]
What are `canmount`, `mountpoint`, and `mounted` set to after that command?

**Reply by knacky34** [reply]
Here is the result of `zfs get mounted,canmount,mountpoint` (slightly reorganized :) )

| dataset | mountpoint | canmount | mounted |
|---|---|---|---|
| dpool | /data | off | no |
| dpool/nextcloud | /data/nextcloud | off | no|
| dpool/nextcloud/data |/data/nextcloud/data|on|no|

**Reply by rincebrain** [reply]
Okay, what does "keystatus" say for them?

**Reply by knacky34** [reply]
@rincebrain, the keystatus is available for all the 3 datasets.

I've setup root on ZFS on Debian 12, so I have a root pool which get decrypted and mounted fine at boot. But I don't if it's related but `dpool` is not detected when running `zfsunlock` from initramfs.

**Comment by knacky34** [comment, marked as answer]
My issue is related to this #12418
Thank you for your help.
